src/shogun/base/SGObject.h

```diff
@@ -774,6 +799,19 @@ class SGObject: public std::enable_shared_from_this<SGObject>
 				);
 			}
 		}
+		else if constexpr (is_auto_value_v<T>)
+		{
+			using ReturnType = traits::get_variant_type_t<0, T>;
+			Any::register_visitor<T, ParameterGetterInterface<ReturnType, T>>(
+				[](T* value, auto* visitor)
+				{
+					if (std::holds_alternative<AutoValueEmpty>(*value))
+						throw std::bad_optional_access{};
```

Comment:
this exception does not have a message, but it is caught in `get` which then throw a ShogunException with meaning

Reply:
maybe add a comment with that?